pkg/cmd/admin/migrate/migrator.go

```diff
@@ -0,0 +1,362 @@
+package migrate
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/kubectl"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+// MigrateVisitFunc is invoked for each returned object, and may return a
+// Reporter that can contain info to be used by save.
+type MigrateVisitFunc func(info *resource.Info) (Reporter, error)
+
+// MigrateActionFunc is expected to persist the altered info.Object. The
+// Reporter returned from Visit is passed to this function and may be used
+// to carry additional information about what to save on an object.
+type MigrateActionFunc func(info *resource.Info, reporter Reporter) error
+
+// Reporter indicates whether a resource requires migration.
+type Reporter interface {
+	// Changed returns true if the resource requires migration.
+	Changed() bool
+}
+
+// ResourceOptions assists in performing migrations on any object that
+// can be retrieved via the API.
+type ResourceOptions struct {
+	In          io.Reader
+	Out, ErrOut io.Writer
+
+	Include       []string
+	Filenames     []string
+	AllNamespaces bool
+	Confirm       bool
+	Output        string
+
+	DefaultExcludes []string
+
+	Builder   *resource.Builder
+	SaveFn    MigrateActionFunc
+	PrintFn   MigrateActionFunc
+	DryRun    bool
+	Summarize bool
+}
+
+func (o *ResourceOptions) Bind(c *cobra.Command) {
+	c.Flags().StringVarP(&o.Output, "output", "o", o.Output, "Output the modified objects instead of saving them, valid values are 'yaml' or 'json'")
+	c.Flags().StringSliceVar(&o.Include, "include", o.Include, "Resource types to migrate. Passing --filename will override this flag.")
+	c.Flags().BoolVar(&o.Confirm, "confirm", false, "If false no migrations will be performed.")
+	c.Flags().BoolVar(&o.AllNamespaces, "all-namespaces", true, "If true, all namespaces will be utilized.")
```

Comment:
Current namespace.  I'll add a comment, but migration is implicitly global,
explicitly namespace scoped.

On Thu, Jun 16, 2016 at 6:02 AM, Maciej Szulik <notifications@github.com>
wrote:

> In pkg/cmd/admin/migrate/migrator.go
> <https://github.com/openshift/origin/pull/9240#discussion_r67318863>:
>
> > +	Output        string
> > +
> > +	DefaultExcludes []string
> > +
> > +	Builder   *resource.Builder
> > +	SaveFn    MigrateActionFunc
> > +	PrintFn   MigrateActionFunc
> > +	DryRun    bool
> > +	Summarize bool
> > +}
> > +
> > +func (o *ResourceOptions) Bind(c *cobra.Command) {
> > +	c.Flags().StringVarP(&o.Output, "output", "o", o.Output, "Output the modified objects instead of saving them, valid values are 'yaml' or 'json'")
> > +	c.Flags().StringSliceVar(&o.Include, "include", o.Include, "Resource types to migrate. Passing --filename will override this flag.")
> > +	c.Flags().BoolVar(&o.Confirm, "confirm", false, "If false no migrations will be performed.")
> > +	c.Flags().BoolVar(&o.AllNamespaces, "all-namespaces", true, "If true, all namespaces will be utilized.")
>
> What if I specify false here, which namespace will be used? It makes more
> sense to either specify --namespace or if none is specified use all.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9240/files/eb1eb8a1f37fe62714438a3a805545a98b82daec..053e3598622ec47ce57195dacc03aedc08e4489f#r67318863>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_pwijBsvCMscBimYUCJhaNin23X34ks5qMR8xgaJpZM4IxnZA>
> .
>